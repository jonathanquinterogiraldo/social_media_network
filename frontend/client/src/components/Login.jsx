import React from "react"
import axios from 'axios'
import { Form, Button, Col, Container, Row } from 'react-bootstrap'
import { useForm } from 'react-hook-form'
import { useHistory } from "react-router-dom"


const apiUrl = process.env.REACT_APP_API_URL

function Login(){

    const history = useHistory();

    const {register, handleSubmit} = useForm();

    const email = register('email',{
        require: true
    })

    const password = register('password',{
        require: true
    })

    const onSubmit = (data) => {
        console.log(data)
        //axios.post(`${apiUrl}/users/login`, data)
        axios({
            method: 'POST',
            baseURL: apiUrl,
            url: '/users/login',
            data: data           
        })        
        .then(
            data => {
                if(data){
                    console.log(data)
                    localStorage.setItem('token', data.data.token)
                    history.push('/home')                      
                }    
            }
        ).catch(
            error => {
                console.log(error)
            }
        )    
    }   
    
    return(      
        <div>     
            <Form onSubmit={handleSubmit(onSubmit)}>
                <Container>
                    <Row >
                        <Col><h3>Inicio de Sesión</h3></Col>
                    </Row>
                </Container>            
                <Form.Group>
                    <Form.Row>
                        <Col sm='auto'>
                            <Form.Label>Correo</Form.Label>
                            <Form.Control type='email'
                                            placeholder='Digite su correo'                                            
                                            onChange={email.onChange}
                                            onBlur={email.onBlur}
                                            name={email.name}
                                            ref={email.ref}/>
                        </Col>
                    </Form.Row>
                </Form.Group>
                <Form.Group>                
                    <Form.Row>
                        <Col sm='auto'>
                            <Form.Label>Contraseña</Form.Label>
                            <Form.Control type='password'
                                           placeholder='Digite su correo'
                                           onChange={password.onChange}
                                           onBlur={password.onBlur}
                                           name={password.name}
                                           ref={password.ref}/>
                        </Col>
                    </Form.Row>
                </Form.Group>
                <Button type='submit'>Enviar</Button>                
            </Form>            
        </div>   
    )
}    

export default Login
