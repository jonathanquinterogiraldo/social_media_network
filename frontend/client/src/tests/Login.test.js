import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import Login from '../components/Login'
import { expect } from '@jest/globals'

describe('Unit Test <Login /> Component', () => {
    test('should mount <Login /> correctly', () => {

        const wrapper = shallow(<Login />)

        expect(wrapper).toMatchSnapshot() 
        
    })   

    test('should show title <Login /> wcomponent ', () => {

        const expectedTitle = 'Inicio de Sesi√≥n'
        const wrapper = shallow(<Login />)
        const title = wrapper.find('h3').text().trim()
        console.log(title)

        expect(title).toBe(expectedTitle)        
    })   
})
