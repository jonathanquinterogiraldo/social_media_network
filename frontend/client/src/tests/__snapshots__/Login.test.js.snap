// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Test <Login /> Component should mount <Login /> correctly 1`] = `
<div>
  <Form
    inline={false}
    onSubmit={[Function]}
  >
    <Container
      fluid={false}
    >
      <Row
        noGutters={false}
      >
        <Col>
          <h3>
            Inicio de Sesión
          </h3>
        </Col>
      </Row>
    </Container>
    <FormGroup>
      <FormRow>
        <Col
          sm="auto"
        >
          <FormLabel
            column={false}
            srOnly={false}
          >
            Correo
          </FormLabel>
          <FormControl
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Digite su correo"
            type="email"
          />
        </Col>
      </FormRow>
    </FormGroup>
    <FormGroup>
      <FormRow>
        <Col
          sm="auto"
        >
          <FormLabel
            column={false}
            srOnly={false}
          >
            Contraseña
          </FormLabel>
          <FormControl
            name="password"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Digite su correo"
            type="password"
          />
        </Col>
      </FormRow>
    </FormGroup>
    <Button
      active={false}
      disabled={false}
      type="submit"
      variant="primary"
    >
      Enviar
    </Button>
  </Form>
</div>
`;
